README
=encoding UTF-8

Module-CPANTS-Analyse version 0.87 is Copyright © 2012, Perl QA team

=head1 NAME

README - basic information for users prior to downloading

=head1 DESCRIPTION

Module-CPANTS-Analyse tests a dist for Kwalitee. This module is part of the
L<CPANTS project|http://cpants.perl.org>.

=head1 AUTHOR

=head2 Distribution maintainer

L<Lars Dɪᴇᴄᴋᴏᴡ|https://metacpan.org/author/daxim>, C<< <daxim@cpan.org> >>

=head2 Contributors

See file F<AUTHORS>.

=head1 COPYRIGHT

Copyright © 2003–2009 L<Thomas Klausner|https://metacpan.org/author/domm>,
C<< <domm@cpan.org> >>

Some files are Copyright © 2006–2008
L<Gábor Szabó|https://metacpan.org/author/szabgab>, C<< <szabgab@gmail.com> >>

Some files are Copyright © 2006
L<Steffen Müller|https://metacpan.org/author/smueller>,
C<< <smueller@cpan.org> >>

Some files are Copyright © 2012
L<Lars Dɪᴇᴄᴋᴏᴡ|https://metacpan.org/author/daxim>, C<< <daxim@cpan.org> >>

The module distributions in the F<t/eg/> directory are copyright by their
respective authors.

=head1 LICENSE

This code is distributed under the same license as Perl.


Changes
=encoding UTF-8

=head1 NAME

Changes - Revision history for Module-CPANTS-Analyse

=head2 0.90_02 2013-08-03

=over

=item Fixed L<rt#87535|https://rt.cpan.org/Ticket/Display.html?id=87535> - incorrect version specification in 0.90_01 (L<ether|https://metacpan.org/author/ether>)

=item Fixed L<rt#87534|https://rt.cpan.org/Ticket/Display.html?id=87534> - Test failure in 0.90_01 (L<ether|https://metacpan.org/author/ether>)

=item Fixed L<rt#87561|https://rt.cpan.org/Ticket/Display.html?id=87561> - t/11_hash_random.t fails due to undeclared test dependency (L<wyant|https://metacpan.org/author/wyant>)

=back

=head2 0.90_01 2013-08-01

=over

=item Applied all the changes in 0.88 again, and removed a few non-portable metrics for Test::Kwalitee.

=back

=head2 0.89 2013-08-01

=over

=item Tentatively reverted all the changes in 0.88

=back

=head2 0.88 2013-08-01

=over

=item add metric C<no_dot_underscore_files>
(L<daxim|https://metacpan.org/author/daxim>,
L<charsbar|https://metacpan.org/author/ishigaki>)

=item remove metrics C<distributed_by_debian>,
C<latest_version_distributed_by_debian>, C<has_no_bugs_reported_in_debian>,
C<has_no_patches_in_debian>, C<no_cpants_errors>, C<uses_test_nowarnings>,
C<has_test_pod>, C<has_test_pod_coverage>, C<has_examples>
(L<charsbar|https://metacpan.org/author/ishigaki>)

=item add metric C<portable_filenames>
(L<charsbar|https://metacpan.org/author/ishigaki>)

=item numerous fixes for a smoother operation of www-cpants
(L<charsbar|https://metacpan.org/author/ishigaki>)

=back

=head2 0.87 2013-02-12

=over

=item fix L<rt#80225|https://rt.cpan.org/Ticket/Display.html?id=80225> - Test
failures due to Test::CPAN::Meta::YAML::Version interface change
(L<wyant|https://metacpan.org/author/wyant>)

=item fix L<rt#82939|https://rt.cpan.org/Ticket/Display.html?id=82939> -
Hash-Randomization causes 10_analyse.t to fail
(L<kentnl|https://metacpan.org/author/kentnl>,
L<chorny|https://metacpan.org/author/chorny>)

=item fix L<rt#44796|https://rt.cpan.org/Ticket/Display.html?id=44796> -
Module::CPANTS::Kwalitee::Manifest broken for MANIFESTs containing files with
spaces (L<rivy|https://metacpan.org/author/rivy>)

=back

=head2 0.86 2012-04-11

=over

=item add several strict and warnings equivalents & make it easy to add more

=item fix when L<Moose> is used and strict is not used

=item add info about C<MIN_PERL_VERSION>

=item better remedy for
L<metayml_declares_perl_version|Module::CPANTS::Kwalitee::MetaYML>

=item L<metayml_declares_perl_version|Module::CPANTS::Kwalitee::MetaYML> moved
from experimental to extra

=item some pod improvements (L<jquelin|https://metacpan.org/author/jquelin>)

=item fix L<rt#65903|https://rt.cpan.org/Ticket/Display.html?id=65903> - no more
C<Test::YAML::Meta::Version> on CPAN
(L<jquelin|https://metacpan.org/author/jquelin>)

=item replace L<YAML::Syck> with L<YAML::Any>
(L<jquelin|https://metacpan.org/author/jquelin>)

=item L<no_symlinks|Module::CPANTS::Kwalitee::Files> checks only files in
F<MANIFEST>, use L<ExtUtils::Manifest/maniread>
(L<rurban|https://metacpan.org/author/rurban>)

=item add more equivalents for L<use_strict|Module::CPANTS::Kwalitee::Uses> and
L<use_warnings|Module::CPANTS::Kwalitee::Uses> tests
(L<doherty|https://metacpan.org/author/doherty>)

=item implement L<valid_signature|Module::CPANTS::Kwalitee::Signature> metric
(L<daxim|https://metacpan.org/author/daxim>)

=back

=head2 0.85 2009-07-12, L<Alexandr Ciornii|https://metacpan.org/author/chorny>

=over

=item fix fails on non-Win32

=back

=head2 0.84 2009-07-08, L<Alexandr Ciornii|https://metacpan.org/author/chorny>

=over

=item require L<Test::Warn> 0.11

=item add L<Test::More> to deps

=item add L<LWP::Simple> to deps

=item fix F<calc_test_yaml_meta.t>

=item better F<META.yml>

=item list of changes for L<0.83|/"0.83 2009-06-10">
(L<domm|https://metacpan.org/author/domm>)

=back

=head2 0.83 2009-06-10

=over

=item ignore C<no_index> when searching for examples
(L<xdaveg|https://metacpan.org/author/dagolden>)

=item skip the debian metrics and downloading the debian file when in LINT
(L<szabgab|https://metacpan.org/author/szabgab>)

=item L<has_test_pod|Module::CPANTS::Kwalitee::Uses> and
L<has_test_pod_coverage|Module::CPANTS::Kwalitee::Uses> are now optional metrics
(L<domm|https://metacpan.org/author/domm>)

=back

=head2 0.82 2008-07-08 10:23:30

=over

=item update F<META.yml>
L<spec to 1.4|http://module-build.sourceforge.net/META-spec-v1.4.html>

=item fixed experimental metrics

=item use L<Moose> now counts for using L<strict>

=item lots of commits from L<Gábor|https://metacpan.org/author/szabgab>

=item honor F<META.yml> key C<no_index>. Fixes
L<RT#32777|https://rt.cpan.org/Ticket/Display.html?id=32777> and an IRC
complaint by L<BinGOs|https://metacpan.org/author/bingos>

=back

=head2 0.81 Sun Apr 13 16:45:29 CEST 2008

=over

=item fixed doc bug in SYNOPSIS (reported by
L<ADAMOWSKI|https://metacpan.org/author/adamowski> as
L<RT#34773|https://rt.cpan.org/Ticket/Display.html?id=34773>)

=item made L<Module::CPANTS::Kwalitee::FindModules> smarter
(L<domm|https://metacpan.org/author/domm>): If available, use F<META.yml>
C<provides>. If not, look for C<package> statements in files. It now also finds
stuff like L<Tk::Widget> and L<App::Ack>

=item upgrade to L<Software::License> 0.003
(L<Gábor|https://metacpan.org/author/szabgab>)

=back

=head2 0.80 Sat Apr 12 10:53:40 CEST 2008

L<domm|https://metacpan.org/author/domm>:

=over

=item add L<metayml_has_provides|Module::CPANTS::Kwalitee::MetaYML> metric

=back

L<Gábor Szabó|https://metacpan.org/author/szabgab>:

=over

=item simplify the way the VERSION number is looked up, keep the full version
line

=item increase the L<no_large_files|Module::CPANTS::Kwalitee::Files> limit to
200_000

=item some unit test fixes thanks to L<Tux|https://metacpan.org/author/hmbrand>

=item some of {error} fields are now ARRAY refs instead of strings

=item more detailed unit tests


Build.PL
use utf8;
use strict;
use Module::Build;

my $build = Module::Build->new(
    dist_author => 'Lars Dɪᴇᴄᴋᴏᴡ <daxim@cpan.org>',
    module_name => 'Module::CPANTS::Analyse',
    license     => 'perl',
    requires    => {
        'Class::Accessor'           => '0.19',
        'Archive::Any::Lite'        => '0.06',
        'Archive::Tar'              => '1.48', #depends on IO::Compress without memory leak
        'CPAN::DistnameInfo'        => '0.06',
        'CPAN::Meta::YAML'          => '0.008',
        'CPAN::Meta::Validator'     => '2.131490',
        'YAML::Any'                 => 0,
        'Module::ExtractUse'        => '0.30', # for used_in_eval
        'Module::Pluggable'         => '2.96',
        'IO::Capture'               => '0.05',
        'Array::Diff'               => '0.04',
        'version'                   => '0.73',
        'File::Find::Rule::VCS'     => '0',
        'Software::License'         => '0.003',
        'perl'                      => 5.006,
        'Set::Scalar'               => 0,
        'Text::Balanced'            => 0,
    },
    build_requires => {
        'Test::More'                => '0',
        'Test::Deep'                => '0',
        'Test::Warn'                => '0.11', #less problems with new Sub::Uplevel
        'Test::NoWarnings'          => '0',
        'Cwd'                       => '0',
    },
    configure_requires => {
        'Module::Build'             => '0.4',
    },
    meta_merge => {
        resources => {
            homepage => 'http://cpants.perl.org',
            repository => 'https://github.com/daxim/Module-CPANTS-Analyse',
            bugtracker => 'https://rt.cpan.org/Public/Dist/Display.html?Name=Module-CPANTS-Analyse',
            },
        keywords => [
            'CPANTS','kwalitee',
        ],
    },
    script_files=>[qw(bin/cpants_lint.pl)],
    create_makefile_pl=>'traditional',
);


$build->add_to_cleanup('Debian_CPANTS.txt');
$build->create_build_script;

